src/utils.js

```diff
@@ -263,3 +263,9 @@ export function formatNoteReferrerEntityData(data) {
     id,
   };
 }
+
+export function eachPromise(arr, fn) {
+  if (!Array.isArray(arr)) return Promise.reject(new Error('Array not found'));
+
+  return arr.reduce((acc, cur) => acc.then(response => fn(response, cur)), Promise.resolve([]));
+}
```

Comment:
The point of using `reduce` rather than `Promise.all` is to cause the fetches to be processed in series rather than in parallel. You can see how it was [used for DELETE and POST](https://github.com/folio-org/ui-users/pull/1476/files) requests in ui-users; maybe we just need to configure the GET mutator correctly -- set `accumulate: true`, I think -- in order to make it return a Promise so we can use [the stripes-util version of `eachPromise`](https://github.com/folio-org/stripes-util/blob/6b8cf3772d82ba4c7a076c099563e00415899c0f/lib/eachPromise.js)?